ext/DataFramesExt.jl

```diff
@@ -0,0 +1,45 @@
+module DataFramesExt
+
+using NEOs
+import NEOs: RadecMPC, RadarJPL
+
+if isdefined(Base, :get_extension)
+    using DataFrames: DataFrame, nrow
+    import Tables: istable, rowaccess, rows, schema, Schema
+else
+    using ..DataFrames: DataFrame, nrow
+    import ..Tables: istable, rowaccess, rows, schema, Schema
+end
+
+# Methods to convert a Vector{RadecMPC{T}} to a DataFrame
+istable(::Type{Vector{RadecMPC{T}}}) where {T <: AbstractFloat} = true
+rowaccess(::Type{Vector{RadecMPC{T}}}) where {T <: AbstractFloat} = true
+rows(x::Vector{RadecMPC{T}}) where {T <: AbstractFloat} = x
+schema(::Vector{RadecMPC{T}}) where {T <: AbstractFloat} = Schema(fieldnames(RadecMPC{T}), Tuple{fieldtypes(RadecMPC{T})...})
+
+# Methods to convert a Vector{RadarJPL{T}} to a DataFrame
+istable(::Type{Vector{RadarJPL{T}}}) where {T <: AbstractFloat} = true
+rowaccess(::Type{Vector{RadarJPL{T}}}) where {T <: AbstractFloat} = true
+rows(x::Vector{RadarJPL{T}}) where {T <: AbstractFloat} = x
+schema(::Vector{RadarJPL{T}}) where {T <: AbstractFloat} = Schema(fieldnames(RadarJPL{T}), Tuple{fieldtypes(RadarJPL{T})...})
+
+# Methods to convert a DataFrame to a Vector{RadecMPC{T}} / Vector{RadarJPL{T}}
+for T in (:(RadecMPC), :(RadarJPL))
+    @eval begin
+        function $T(df::DataFrame)
+            @assert all(string.(fieldnames($T)) .== names(df))
+            coltypes = eltype.(eachcol(df))
+            numtypes = filter!(x -> x <: AbstractFloat, coltypes)
+            @assert all(numtypes .== numtypes[1])
+            S = numtypes[1]
+            obs = Vector{$T{S}}(undef, nrow(df)) 
+            for i in eachindex(obs)
+                obs[i] = $T(values(df[i, :])...)
+            end 
+
+            return obs 
+        end 
+    end
```

Comment:
```suggestion
# Methods to convert a DataFrame to a Vector{RadecMPC{T}} / Vector{RadarJPL{T}}
function Vector{T}(df::DataFrame) where {T <: AbstractAstrometry}
    @assert all(string.(fieldnames(T)) .== names(df))
    coltypes = eltype.(eachcol(df))
    numtypes = filter!(x -> x <: AbstractFloat, coltypes)
    @assert all(numtypes .== numtypes[1])
    obs = Vector{T}(undef, nrow(df))
    for (i, row) in zip(eachindex(obs), eachrow(df))
        obs[i] = T(values(row)...)
    end

    return obs
end

convert(::Type{Vector{T}}, df::DataFrame) where {T <: AbstractAstrometry} = Vector{T}(df::DataFrame)
```

Reply:
After thinking more about it, I would suggest (see suggestion) to define a `Vector{T} where T<:AbstractAstrometry` constructor, which I think it's conceptually clearer and avoids having duplicate methods which do essentially the same for both types of astrometry. I also suggest to add a `convert` method, which would allow julia to find the appropriate constructor, though I'm not 100% sure this last thing is needed. If we go for this route we would have to import Base.convert in the extension, and update the tests so that we construct from a DataFrame as `Vector{RadecMPC}(df)`.